### DISCUSSION TITLE
Example of testing endpoints and islands?

### DISCUSSION THREAD

**Discussion by cj** [freshframework/fresh#300] [Q&A] [answered]
Hi,

Is there an example of testing endpoints or islands?

Cheers!

**Comment by lucacasonato** [comment, marked as answer]
Testing routes: https://github.com/denoland/fresh/blob/main/tests/main_test.ts#L3-L38

Testing islands: https://github.com/denoland/fresh/blob/main/tests/islands_test.ts

**Reply by cj** [reply]
Thank you!

**Comment by arvigeus** [comment]
Can this be done without puppeteer? I was trying to make it work with `@testing-library/preact`, but I only get `Cannot read properties of undefined (reading '__H')` (using [this](https://www.sitepen.com/blog/doing-it-all-with-deno) as guide)

**Reply by lucacasonato** [reply]
Make sure to import `https://esm.sh/@testing-library/preact?deps=preact@10.8.1` rather than `https://esm.sh/@testing-library/preact` (specify the exact preact version you are using).

**Reply by arvigeus** [reply]
It works, awesome! I hope to see testings as first class citizen in this stack as well.
